client/app/visualizations/chart/plotly/index.js

```diff
@@ -34,6 +40,10 @@ const PlotlyChart = () => ({
     let layout = {};
     let data = [];
 
+    const tooltip = $('<div>')
+      .addClass('plotly-chart-tooltip')
+      .appendTo('body');
```

Comment:
So we render our own tooltip... I can see how this can become a maintenance nightmare, chasing various Plotly edge cases 🤔 

What do we lose if we go with Plotly's builtin tooltip?

Reply:
Past experience shows that there are always unexpected scenarios with charts :) Once we merge and deploy this, we're committed to it. But I don't think we need this distraction right now (having to maintain and fix issues related to the tooltip) -- we have more interesting stuff to do.

So let's shelf this for now, and implement something simpler:

* First step: expose the ability to set `hoverformat` for the X and Y axes.
* Second step: feature to set the `hovertext` property using data from the current point.

While it's not as complete as what you implemented here, it will serve most of the cases.